projects/keycloak-angular/src/lib/core/services/keycloak.service.spec.ts

```diff
@@ -63,5 +63,33 @@ describe('KeycloakService', () => {
         expect(excludedRegex2.httpMethods[0]).toBe('GET');
       }
     ));
+
+    it('Should not call login in error case if getToken is called with forceLogin false', inject(
```

Comment:
Thank you for adding tests to your PR it's much appreciated as our coverage is not very good at the moment. This test looks good to me but I'd also like to add a test here to see that `login` is indeed called if `forceLogin` is not specified.

Reply:
I tried to do that but with the current implementation it is not possible.
To assert that `login` is called it would be necessary to call `await getToken()`. But with the current implementation `getToken()` never resolves when `this.login()` is called. Therefore it is not possible to await the `getToken()` call which makes it impossible to write this test.

If used in a real scenario this is unproblematically because `login()` performs a redirect anyway.
A solution to this would be to change the line `this.login()` to `return this.login()` in the `getToken` function. 
I decided against doing this, because it changes the behaviour of `getToken()` and I did'n want to make a possibly breaking change.

Nevertheless, in my opinion the current behaviour of `getToken` is wrong and we should change it.
If you want, I can implement the suggested changes and add the test case you wanted as well.

Reply:
I wonder if the redirect is performed also when `silentCheckSsoRedirectUri` is enabled. Let me do some investigation as well, I agree with you that this implementation should probably be changed as well.